src/prefect/projects/steps/pull.py

```diff
@@ -27,6 +28,107 @@ def set_working_directory(directory: str) -> dict:
+        pull:
+            - prefect.projects.steps.git_clone:
+                repository: https://github.com/PrefectHQ/prefect.git
+        ```
+
+        Clone a branch of a public repository:
+        ```yaml
+        pull:
+            - prefect.projects.steps.git_clone:
+                repository: https://github.com/PrefectHQ/prefect.git
+                branch: my-branch
+        ```
+
+        Clone a private repository using an access token:
+        ```yaml
+        pull:
+            - prefect.projects.steps.git_clone:
+                repository: https://github.com/org/repo.git
+                access_token: "{{ prefect.blocks.secret.github-access-token }}" # Requires creation of a Secret block
+        ```
+        Note that you will need to [create a Secret block](/concepts/blocks/#using-existing-block-types) to store the
+        value of your git credentials. You can also store a username/password combo or token prefix (e.g. `x-token-auth`)
+        in your secret block. Refer to your git providers documentation for the correct authentication schema.
+
+        Clone a repository with submodules:
+        ```yaml
+        pull:
+            - prefect.projects.steps.git_clone:
+                repository: https://github.com/org/repo.git
+                include_submodules: true
+        ```
+
+        Clone a repository with an SSH key (note that the SSH key must be added to the worker
+        before executing flows):
+        ```yaml
+        pull:
+            - prefect.projects.steps.git_clone:
+                repository: git@github.com:org/repo.git
+        ```
+    """
+    url_components = urllib.parse.urlparse(repository)
+    if url_components.scheme == "https" and access_token is not None:
+        updated_components = url_components._replace(
+            netloc=f"{access_token}@{url_components.netloc}"
+        )
+        repository_url = urllib.parse.urlunparse(updated_components)
+    else:
+        repository_url = repository
+
+    cmd = ["git", "clone", repository_url]
+    if branch:
+        cmd += ["-b", branch]
+    if include_submodules:
+        cmd += ["--recurse-submodules"]
+
+    # Limit git history
+    cmd += ["--depth", "1"]
+
+    try:
+        subprocess.check_call(
+            cmd, shell=sys.platform == "win32", stderr=sys.stderr, stdout=sys.stdout
+        )
+    except subprocess.CalledProcessError as exc:
+        # Hide the command used to avoid leaking the access token
+        exc_chain = None if access_token else exc
+        raise RuntimeError(
+            f"Failed to clone repository {repository!r} with exit code"
+            f" {exc.returncode}."
+        ) from exc_chain
+
+    directory = "/".join(repository.strip().split("/")[-1:]).replace(".git", "")
+    projects_logger.info(f"Cloned repository {repository!r} into {directory!r}")
+    return {"directory": directory}
+
+
+@deprecated_callable(start_date="Jun 2023", help="Use 'git clone' instead.")
 def git_clone_project(
```

Comment:
For this deprecated function, I recommend updating the docstring so that it's clear that it's deprecated in the docs. Also, we can update this as pass through to the new `git_clone` function so that if we make an update to `git_clone`, we don't need to remember to port the change back here.

Somthing like this:
```python
def git_clone_project(*args, **kwargs):
    """Deprecated. Use `git_clone` instead."""
    git_clone(*args, **kwargs)
```

Reply:
Great idea!